single_kernel_mongo/core/secrets.py

```diff
@@ -21,14 +21,14 @@
 logger = logging.getLogger(__name__)
 
 
-def generate_secret_label(app_name: str, rel_name: str, scope: Scope) -> str:
+def generate_secret_label(app_name: str, scope: Scope) -> str:
     """Generate unique group_mappings for secrets within a relation context.
 
     Defined as a standalone function, as the choice on secret labels definition belongs to the
     Application Logic. To be kept separate from classes below, which are simply to provide a
     (smart) abstraction layer above Juju Secrets.
     """
-    members = [rel_name, app_name, scope.value]
+    members = [app_name, scope.value]
```

Comment:
will this work in the context of a cross model relation were multiple apps with the same name may reside in multiple models? or what was the original reason for adding the relation name to the label? 
also, would this break backwards compatibility with the current stable release?

Reply:
I think that it works in both cases because the `rel_name` here would be `database-peers` or `router-peers`. Those secrets are used only by the application in the scope of peer relations and never shared so I think they should be local to this application/to this model.

I had to patch that because it broke because the current secrets labels are created with `app_name` and `scope` but not the relation, so on upgrade everything would break because of that (no access to secrets at all)